lib/web/oidcidp.go

```diff
@@ -89,6 +93,61 @@ func (h *Handler) jwksOIDC(_ http.ResponseWriter, r *http.Request, _ httprouter.
 	return &resp, nil
 }
 
+// thumbprint returns the thumbprint as required by AWS when adding an OIDC Identity Provider.
+// This is documented here:
+// https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+func (h *Handler) thumbprint(_ http.ResponseWriter, r *http.Request, _ httprouter.Params) (interface{}, error) {
+	// Dial requires the following address format: host:port
+	cfgPublicAddress := h.cfg.PublicProxyAddr
+	if !strings.Contains(cfgPublicAddress, "://") {
+		cfgPublicAddress = "https://" + cfgPublicAddress
+	}
+
+	addrURL, err := url.Parse(cfgPublicAddress)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	port := addrURL.Port()
+	if port == "" {
+		port = "443"
```

Comment:
wouldn't the default be 3080 if nothing else has been set?

Reply:
It should be
I was being extra cautious but I think we can remove this default